### PR TITLE
chore: add go entry point for `snyk code test`

### PR DESCRIPTION
**Pull request by thisislawatts**
## Pull Request Submission

Please check the boxes once done.

The pull request must:

- **Reviewer Documentation**
    - [ ] follow [CONTRIBUTING](https://github.com/snyk/cli/blob/main/CONTRIBUTING.md) rules
    - [ ] be accompanied by a detailed description of the changes
    - [ ] contain a risk assessment of the change (Low | Medium | High) with regards to breaking existing functionality. A change e.g. of an underlying language plugin can completely break the functionality for that language, but appearing as only a version change in the dependencies.
    - [ ] highlight breaking API if applicable
    - [ ] contain a link to the automatic tests that cover the updated functionality.
    - [ ] contain testing instructions in case that the reviewer wants to manual verify as well, to add to the manual testing done by the author.
    - [ ] link to the link to the PR for the User-facing documentation
- **User facing Documentation**
    - [ ] update any relevant documentation in gitbook by submitting a gitbook PR, and including the PR link here
    - [ ] ensure that the message of the final single commit is descriptive and prefixed with either `feat:` or `fix:` , others might be used in rare occasions as well, if there is no need to document the changes in the release notes. The changes or fixes should be described in detail in the commit message for the changelog & release notes.
- **Testing**
    - [ ] Changes, removals and additions to functionality must be covered by acceptance / integration tests or smoke tests - either already existing ones, or new ones, created by the author of the PR.

## Pull Request Review

All pull requests must undergo a thorough review process before being merged. 
The review process of the code PR should include code review, testing, and any necessary feedback or revisions. 
Pull request reviews of functionality developed in other teams only review the given documentation and test reports. 

Manual testing will not be performed by the reviewing team, and is the responsibility of the author of the PR.

For Node projects: It’s important to make sure changes in `package.json` are also affecting `package-lock.json` correctly.

****************************If a dependency is not necessary, don’t add it.**************************** 

When adding a new package as a dependency, make sure that the change is absolutely necessary. We would like to refrain from adding new dependencies when possible.
Documentation PRs in gitbook are reviewed by Snyk's content team. They will also advise on the best phrasing and structuring if needed.

## Pull Request Approval

Once a pull request has been reviewed and all necessary revisions have been made, it is approved for merging into 
the main codebase. The merging of the code PR is performed by the code owners, the merging of the documentation PR 
by our content writers.

## What does this PR do?

Removes strict validation for the `snyk code test` workflow. At the moment if any flags/arguments are provided which do not meet the defined guidelines the command will fail. This risks introducing a breaking change for customers. To reduce this risk we are introducing a branch to our flag validation which is less strict for the new `snyk code` workflow. 

## How should this be manually tested?

```
make build && ./binary-releases/snyk-macos-arm64 code test --fake-flag-which-is-definitely-not-expeceted
```

[CLI-187]

[CLI-187]: https://snyksec.atlassian.net/browse/CLI-187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cliv2/cmd/cliv2/main.go [hunk 1]
```diff
@@ -259,6 +259,11 @@ func createCommandsForWorkflows(rootCommand *cobra.Command, engine workflow.Engi
 		parentCommand.RunE = runCommand
 		parentCommand.Hidden = !workflowEntry.IsVisible()
 		parentCommand.DisableFlagParsing = false
+
+		// special case for snyk code test, to preserve backwards compatibility we will need to relax flag validation
```
Comment (PeterSchafer) [inline cliv2/cmd/cliv2/main.go:263]: Suggestion: let's create a ticket to keep track of this tech debt. I assume we either remove this or replace it by a more generic approach at some point in the future.

File: test/jest/acceptance/snyk-code/snyk-code.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { runSnykCLI } from '../../util/runSnykCLI';
+
+describe('code', () => {
```
Comment (PeterSchafer) [inline test/jest/acceptance/snyk-code/snyk-code.spec.ts:3]: Suggestion: tests that expect certain text might increase maintenance efforts when the texts are changed. How about running `snyk code test --help` once on purpose and then comparing it against the output in the different test cases.

File: cliv2/cmd/cliv2/main.go (hunk 1 above)
Comment (thisislawatts) [reply cliv2/cmd/cliv2/main.go:263]: https://snyksec.atlassian.net/browse/CLI-237

File: test/jest/acceptance/snyk-code/snyk-code.spec.ts (hunk 2 above)
Comment (thisislawatts) [reply test/jest/acceptance/snyk-code/snyk-code.spec.ts:3]: Updated: https://github.com/snyk/cli/pull/5121/commits/54b89a3a5b133cca24491c5c6f59267b8d946191

**Review by PeterSchafer — APPROVED** [review]

**Pull request merged**